app/routes/join.js

```diff
@@ -1,3 +1,6 @@
 import Route from '@ember/routing/route';
+import { inject as service } from '@ember/service';
 
-export default class JoinRoute extends Route {}
+export default class JoinRoute extends Route {
+  @service login;
```

Comment:
Why you calling service login here??

Reply:
Because to access the role user must be login